templates/keepalived.conf.j2

```diff
@@ -84,6 +84,9 @@ vrrp_instance {{ name }} {
   {% if instance.preempt_delay is defined %}
   preempt_delay {{ instance.preempt_delay }}       # Seconds after startup until preemption. 0 (default) to 1,000.
   {% endif %}
+  {% if instance.advert_int is defined and instance.advert_int | int %}
```

Comment:
anything real number between 0 and 1 will be ignored, as it will be floored to 0. I am not sure if that's you want, as the value will basically not be set.

I would rather have an explicit failure by removing ``and instance.advert_int | int`` in the condition.

Reply:
That option will simply be not set if it's not an INT and keepalived will use the default `advert_int 1`. 
But I can change that. no problem.

Reply:
alright it's just an converter with default behavior and it will not fail in that stage

https://jinja.palletsprojects.com/en/2.10.x/templates/#int
>Convert the value into an integer. If the conversion doesn’t work it will return 0. 

for example:
with the following template
```    {% if rserver.weight is defined %}
    weight {{ rserver.weight | int }}
    {% endif %}
    {% if rserver.uthreshold is defined %}
    uthreshold {{ rserver.uthreshold | int }}
    {% endif %}
    {% if rserver.lthreshold is defined %}
    lthreshold {{ rserver.lthreshold | int }}
    {% endif %}```

```     - ip: 10.0.0.1
        port: 5445
        weight: 0.2
        uthreshold: blubber
        lthreshold: 12
        tcp_checks:
          - connect_port: 5445
            connect_timeout: 5```

will result in 
```  real_server 10.0.0.1 5445 {
    weight 0
    uthreshold 0
    lthreshold 0
    TCP_CHECK {
      connect_port 5445
      connect_timeout 5
    }```

not really what we want.
the same results with the `advert_int`
so I would suggest to keep the templates as is, with `and instance.advert_int | int` and simply not set it if it's not a correct INT and use the default upstream settings instead.
Same for `weight`, `uthreshold` and `lthreshold`
We could set it to other default values for example `| int(1)` but that stuff needs to be updated if upstream is changing it.

If you would like to fail if it's not an INT, it should be done in `main.yml` ```- name: Configure keepalived
  template:
    src: keepalived.conf.j2
    dest: "{{ keepalived_config_file_path }}"
  tags:
    - keepalived-config
  notify:
    - restart keepalived```
either as an extra task or via `when`
Opinions?

Reply:
I thought the int in advert_int stands for interval, which didn't need to be an integer. In this case, anything works, including for example 0.82.

So therefore the condition on line 87 should not test or cast into an int (and therefore should be ``{% if instance.advert_int is defined %}``)

The advert_int line 88 is right the way it is right now (``advert_int {{ instance.advert_int }}``).